internal/voting/sql.go

```diff
@@ -105,6 +105,12 @@ const (
 
 	// containsBody checks if a resolution has a body
 	containsBody = `SELECT body is not null FROM ` + votingSchemaName + `.resolutions WHERE id = $1;`
+	// containsBodyArray checks if a set of resolutions have bodies.
+	// it will return as many rows as the input array has elements, with a boolean for each.
+	containsBodyArray = `WITH input AS (SELECT unnest($1::BYTEA[]) AS id)
```

Comment:
This is good for the contract of `ResolutionsContainsBody`, but you're ultimately after `doesNotHaveBody` in `ProposerTxs`, so a left join (and the subquery with unnest) looks to be too much for the job (we want a set of ids that have body, a subset of the inputs with no real need for the false entries since that is implied by their absence).

`SELECT id FROM kwild_voting.resolutions WHERE id =ANY($1)`

and `ResolutionsContainsBody` would just be doing `idWithBody := row[0].([]byte)` in the rows loop, and the app would do the matching to create the `[]bool`.

What you have works, but I wanted to point out the simpler SQL approach in case it wasn't apparent.  It's more testable and less error prone, less at the mercy of quirks and guarantees of postgres in handling of nulls, defaults, and ordering with array-to-rows etc.

Reply:
Ah that's a good point. I think we maybe just leave this for now, simply because the first thing I am going to do after this release is address the unnecessarily inefficient access patterns around events and voting. Is that ok with you?

Reply:
Never mind, I think failing CI is actually related to the ordering of the booleans so I am going to address it now